openedx/features/edly/api/v1/views/enrollment_count.py

```diff
@@ -0,0 +1,67 @@
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import status, viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from common.djangoapps.student.models import CourseAccessRole, CourseEnrollment
+from openedx.core.djangoapps.catalog.utils import get_programs
+from openedx.features.edly.permissions import IsWpAdminUser
+from openedx.features.edly.utils import is_course_org_same_as_site_org
+
+
+class EdlyProgramEnrollmentCountViewSet(viewsets.ViewSet):
+    """
+    **Use Case**
+
+        Get the number of enrollments for a program.
+
+    **Example Request**
+
+        GET /api/v1/programs/enrollment_count/{program_uuid}
+
+    **GET Parameters**
+
+        * program_uuid: The UUID of the program.
+
+    **Response Values**
+
+        If the request is successful, the request returns an HTTP 200 "OK" response.
+
+        The HTTP 200 response has the following values.
+
+        * enrolled_learners_count: The number of enrollments for the program.
+    """
+    permission_classes = (IsAuthenticated, IsWpAdminUser)
+
+    def retrieve(self, request, pk=None):
+        """
+        Get the number of enrolled learners in a given program.
+        """
+        program = get_programs(uuid=pk)
+        if not program:
+            return Response({'detail': 'Program not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        course_run_ids = [
+            course_run['key']
+            for course in program.get('courses', [])
+            for course_run in course.get('course_runs', [])
+        ]
```

Comment:
Extract logic for fetching course run IDs in a separate function (it could also be a util)